src/core/lib/surface/call.cc

```diff
@@ -269,6 +276,7 @@ class Call : public CppImplOf<Call, grpc_call> {
   mutable Mutex peer_mu_;
   Slice peer_string_;
   gpr_cycle_counter start_time_ = gpr_get_cycle_counter();
+  Timestamp start_timestamp_ = Timestamp::Now();
```

Comment:
Seems like storing this timestamp in two different forms would increase per-call memory, which seems undesirable.  Do we really not have an alternative here?

Reply:
I was thinking that maybe the cycle counter is useful for profiling, but maybe that is not enough of a justification to increase the call size. I think I'll just save a `gpr_timespec`

Reply:
Changed

Reply:
Just to note:

```
sizeof(Timestamp)==8
sizeof(gpr_cycle_counter)==8
sizeof(gpr_timespec)==16
```

But I'm supportive of getting this down to one time query!

(and I think the ultimate fix here is to move `grpc_core::Timestamp` to be nanosecond precise)

Reply:
If `gpr_timespec` is larger, switching to that type is still a regression.

If we change `grpc_core::Timestamp` to nanosecond precision, would that change its size?  If not, maybe we should just go ahead and do that.

Are we worried about latency calculations using `Timestamp` instead of `gpr_cycle_counter` in terms of NTP clock smearing?

Reply:
`Timestamp` is monotonic, so no NTP clock interactions.

I'm not sure that we've done much deep thinking about storage format for nanosecond sized times.

One option would be what absl/gpr_timespec do: a seconds `uint64_t` and a microseconds `uint32_t` - this would increase the size of `Timestamp` everywhere but have some nice compatibility story.

Another would be integer nanoseconds from monotonic clock epoch - which has a maximum range of about 500 years (and worst case starting time of Jan 1 1970 I guess? -- so we'd get about 448 more years).

Another would be floating point nanoseconds from monotonic clock epoch - makes math marginally easier - and probably gets us the precision we need, but we'd need to study a bit.

Reply:
> Another would be integer nanoseconds from monotonic clock epoch - which has a maximum range of about 500 years (and worst case starting time of Jan 1 1970 I guess? -- so we'd get about 448 more years).

I think this is fine. We could probably also consider microsecond? 

> Another would be floating point nanoseconds from monotonic clock epoch - makes math marginally easier - and probably gets us the precision we need, but we'd need to study a bit.

I don't see the point of using floating point for time. We'd be losing precision compared to `uint64_t`. (The precision for 64 bit floating point is 53 bits.)